src/renderer/views/Pins.js

```diff
@@ -54,14 +56,48 @@ class Pins extends Component {
     constructor(props) {
         super(props);
         this.state = {
-            pinls: []
+            pinls: [],
+            newVideos: [],
+            trendingVideos: []
         }
         this.pid = null;
         this.generate = this.generate.bind(this)
     }
     async componentDidMount() {
         await this.generate();
         this.pid = setInterval(this.generate, 1500)
+        this.updateSearchTables()
+    }
+    updateSearchTables( community = null ) {
+        let ids = this.state.pinls.map(x => {return x._id})
+        console.log(ids)
+        let newUrl = 'https://3speak.co/apiv2/feeds/new?limit=10'
+        let trendingUrl = 'https://3speak.co/apiv2/feeds/trending?limit=10'
+        if (community) {
+            newUrl = `https://3speak.co/apiv2/feeds/community/${community}/new?limit=10`
+            trendingUrl = `https://3speak.co/apiv2/feeds/community/${community}/trending?limit=10`
+        }
+        fetch(newUrl)
+            .then(r => r.json())
+            .then(r => {
```

Comment:
Need to have a filter here to ensure with 100% accuracy the videos are indeed available on IPFS, and fetch more videos to fill in the 10 slots with all IPFS videos... We are close to 100% of videos, but it would be nice to have that guarantee especially on the pins page. This is also important when looking on communities that may or may not have very many IPFS videos.